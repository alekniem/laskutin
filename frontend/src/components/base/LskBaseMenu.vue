<template>
    <v-menu bottom left>
        <template #activator="{ attrs, on }">
            <v-btn :id="activatorId" icon v-bind="attrs" v-on="on">
                <v-icon>mdi-dots-vertical</v-icon>
            </v-btn>
        </template>
        <v-list>
            <slot />
        </v-list>
    </v-menu>
</template>

<script>
export default {
    name: "LskBaseMenu",

    props: {
        activatorId: { type: String, required: true },
    },
};
</script>
