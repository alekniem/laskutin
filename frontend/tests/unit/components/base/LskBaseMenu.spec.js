import flushPromises from "flush-promises";
import LskBaseMenu from "@/components/base/LskBaseMenu";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseMenu", () => {
    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskBaseMenu, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
            },
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("no content in default slot, menu hidden");
    });

    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskBaseMenu, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
            },
        });
        await flushPromises();
        wrapper.find("button").trigger("click");
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("no content in default slot, menu visible");
    });

    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskBaseMenu, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
            },
            slots: {
                default: "<div>DEFAULT</div>",
            },
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("content in default slot, menu hidden");
    });

    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskBaseMenu, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
            },
            slots: {
                default: "<div>DEFAULT</div>",
            },
        });
        await flushPromises();
        wrapper.find("button").trigger("click");
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("content in default slot, menu visible");
    });
});
