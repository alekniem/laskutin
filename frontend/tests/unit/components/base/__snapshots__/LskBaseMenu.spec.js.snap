// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseMenu should match snapshot: content in default slot, menu hidden 1`] = `
<div class="v-menu">
  <button type="button"
          class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
          id="TEST_ACTIVATOR_ID"
          role="button"
          aria-haspopup="true"
          aria-expanded="false"
  >
    <span class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-dots-vertical theme--light"
      >
      </i>
    </span>
  </button>
</div>
`;

exports[`LskBaseMenu should match snapshot: content in default slot, menu visible 1`] = `
<div class="v-menu">
  <button type="button"
          class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
          id="TEST_ACTIVATOR_ID"
          role="button"
          aria-haspopup="true"
          aria-expanded="true"
  >
    <span class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-dots-vertical theme--light"
      >
      </i>
    </span>
  </button>
  <transition-stub name="v-menu-transition">
    <div role="menu"
         class="v-menu__content theme--light menuable__content__active "
         style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 8; display: none;"
    >
      <div class="v-list v-sheet theme--light">
        <div>
          DEFAULT
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`LskBaseMenu should match snapshot: no content in default slot, menu hidden 1`] = `
<div class="v-menu">
  <button type="button"
          class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
          id="TEST_ACTIVATOR_ID"
          role="button"
          aria-haspopup="true"
          aria-expanded="false"
  >
    <span class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-dots-vertical theme--light"
      >
      </i>
    </span>
  </button>
</div>
`;

exports[`LskBaseMenu should match snapshot: no content in default slot, menu visible 1`] = `
<div class="v-menu">
  <button type="button"
          class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
          id="TEST_ACTIVATOR_ID"
          role="button"
          aria-haspopup="true"
          aria-expanded="true"
  >
    <span class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-dots-vertical theme--light"
      >
      </i>
    </span>
  </button>
  <transition-stub name="v-menu-transition">
    <div role="menu"
         class="v-menu__content theme--light menuable__content__active "
         style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 8; display: none;"
    >
      <div class="v-list v-sheet theme--light">
      </div>
    </div>
  </transition-stub>
</div>
`;
